cadl-tests/cadl/lro.tsp

```diff
@@ -15,12 +15,13 @@ using Azure.Core.Traits;
 @useDependency(Azure.Core.Versions.v1_0_Preview_2)
 namespace Cadl.LongRunning;
 
+@resource("resources")
 model Resource {
   @visibility("read")
   id: string;
 
   @key
-  @segment("resources")
+  @visibility("create", "update")
```

Comment:
is `segement` be deprecated?

Reply:
I thought previous `@segment` actually means `name` is read-only in JSON.

It returns in response, but not settable in request (as name is set to path, not JSON).

Reply:
`@segment` will be inserted into the operation route before the path parameter's name field.
https://microsoft.github.io/typespec/standard-library/rest/reference/decorators#@TypeSpec.Rest.segment
`@segment` is not deprecated.

tsp compiler throw error when a property defines a key, but does not define visiblity. 
It also reports error if we don't set `@segment` or `@resource`. But it recommends using `@resource` in error message.
```
Model type 'Cadl.LongRunning.Resource' is not a valid resource type. It must be decorated with the '@resource' decorator.
```
[playground](https://cadlplayground.z22.web.core.windows.net/cadl-azure/?c=aW1wb3J0ICJAdHlwZXNwZWMvcmVzdCI7DQrJGmF6dXJlLXRvb2xzL8gmLcYVY29yZcQsDQp1c2luZyBUeXBlU3BlYy5IdHRwO9EWUmVzdNIWVmVyc2lvbmluZ8kcQcReLkNvcmXTEy5Gb3VuZGF0aW9uc9QfVHJhaXTEGg0KQHNlcnZpY2Uoew0KICB0aXRsZTogIkxvbmdSdW7EciIsxBl25gCCOiAiMS4wLjDEFX0pDQpAdXNlRGVwZW5kZW5jeSjLY%2BcAs3MudjFfMF9QcmV2aWV3XzIpDQpuYW1lc3BhY2UgQ2FkbC7La%2BUAkQ0KbW9kZWwgUmVzb3VyY2Ug5QCZQHZpc2liaWxpdHkoInJlYWQiKcQXaWQ6IHN0csg8ICBAa2V50TBjcmVhdGUiLCAidXBkxArFPOQAiMs%2BICDkAbrLEX3qAIxFeOQB6VBhcmFtc%2BcAkHF1ZXLFanByb2plY3RGaWxl5wD03ERlZO4A2O0AwSAgcscdVXJp0EZpbnRlcmbkATfIQk9wZXLmAeTEO2V4dGVuZHPsAefSKTxOb1JlcGVhdGFibGVSZXF1ZXN0cyAmIE5vQ29uZGnEJGFszhhsaWVudMcTSWQ%2BIHvlAI5Acm91dGUoIi9sb25nLXLnAisp7ACn6wHZT3DmAO7mAYlPclXlAYcgaXPzAMzpALZDzS08yBc%2B6AHqcG9sbMVcxzoo00xHZXTRFlN0YXR1c8lQLCBuZXZlcj7lAhToAKFSZXBs5ADE9gCi6wDa8ACtxznxAK5nZXTXTcgTUmVhZP8A5f8A5f8A5ekA5WRlbGX5AX7zANxExTD%2FAJz%2FAJz9AJzwAxvmAKdlxRb%2FAKfrAKdBY8Qb0VfmA7bTZeQDWQ%3D%3D)

Reply:
OK, what about "read" vs. "create", "update"?

Reply:
Yeah, in this case `read` would be better.

Reply:
OK, could you use `read`?

Wondering why this does not affect generated code>?

Reply:
I will update.